beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -128,24 +126,17 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
         target_head_slot: Slot,
         target_head_root: Hash256,
         peer_id: PeerId,
-        is_finalized_segment: bool,
         log: &slog::Logger,
     ) -> Self {
         let mut peers = FnvHashMap::default();
         peers.insert(peer_id, Default::default());
 
         let id = SyncingChain::<T>::id(&target_head_root, &target_head_slot);
 
-        let target_slot = if is_finalized_segment {
-            target_head_slot + (2 * T::EthSpec::slots_per_epoch()) + 1
-        } else {
-            target_head_slot
-        };
-
```

Comment:
This logic exists on Deneb branch from #3827, need to confirm if this is necessary

Reply:
if it's worth anything, agree with Sean that this is the way to go. If we forgo this logic sync will be trapped in a loop because syncing up to X doesn't finalize the chain locally, as the comment says

Reply:
Got it! Thanks for the explanation :) Updated in https://github.com/sigp/lighthouse/pull/4432/commits/d1146ec8b580f8c8c8a6fe0d0b60b63196f87637